hazelcast/src/hazelcast/cp/cp_impl.cpp

```diff
@@ -80,18 +80,15 @@ proxy_session_manager::create_new_session(const raft_group_id& group_id)
 {
     // the lock_ is already acquired as write lock
     auto response = request_new_session(group_id);
-    auto result = sessions_.emplace(
-      group_id, session_state{ response.id, response.ttl_millis });
-    if(!result.second){
-        sessions_.erase(result.first);
-        result = sessions_.emplace(group_id, session_state{response.id, response.ttl_millis});
-    }
-    auto session =
-      sessions_
-        .emplace(group_id, session_state{ response.id, response.ttl_millis })
-        .first;
+    session_state state { response.id , response.ttl_millis };
+
+    auto result = sessions_.emplace( group_id, state );
+
+    if(!result.second)
+        result.first->second = state;
```

Comment:
```suggestion
    if(!result.second) {
        result.first->second = session_state state { response.id , response.ttl_millis };
   }
```
Can you use move operator and add move assignment operator instead?